recipes/opencv/4.x/conanfile.py

```diff
@@ -77,12 +89,20 @@ def configure(self):
             raise ConanInvalidConfiguration("Clang 3.x can build OpenCV 4.x due an internal bug.")
         if self.options.shared:
             del self.options.fPIC
+        if self.options.with_cuda:
+            self.options.contrib = True  # contribs are needed for CUDA support
+        if not self.options.contrib:
+            self.options.with_freetype = False
+            self.options.with_harfbuzz = False
+            self.options.with_glog = False
+            self.options.with_gflags = False
```

Comment:
don't set options in `configure()`

for cuda:
raise in `configure()` if contrib = False

for others options:
delete those options in `configure()` if contrib = False, and use self.options.get_safe in `requirements()`, `build()`, `package_info()` etc